api/src/main/java/javax/jdo/Query.java

```diff
@@ -759,18 +759,19 @@ void addSubquery(
   Integer getDatastoreWriteTimeoutMillis();
 
   /**
-   * Method to cancel any executing queries. If the underlying datastore doesn't support
-   * cancellation of queries this will throw JDOUnsupportedOptionException. If the cancellation
-   * fails (e.g in the underlying datastore) then this will throw a JDOException.
+   * Method to cancel all executions of this query instance. If the underlying datastore doesn't
+   * support cancellation of queries this will throw JDOUnsupportedOptionException. If the
+   * cancellation fails (e.g in the underlying datastore) then this will throw a JDOException.
    *
    * @since 3.0
    */
   void cancelAll();
 
   /**
-   * Method to cancel an executing query in the specified thread. If the underlying datastore
-   * doesn't support cancellation of queries this will throw JDOUnsupportedOptionException. If the
-   * cancellation fails (e.g in the underlying datastore) then this will throw a JDOException.
+   * Method to cancel any executions of this query instance that run in the specified thread. If the
+   * underlying datastore doesn't support cancellation of queries this will throw
+   * JDOUnsupportedOptionException. If the cancellation fails (e.g in the underlying datastore) then
+   * this will throw a JDOException.
    *
    * @param thread The thread to cancel
```

Comment:
I believe that there is no way for a single thread to have multiple simultaneous executions of a query. So there is no reason to change the text of the javadoc for cancelAll.

Reply:
Even so I think it would still be more clear to say “of this query instance” instead of “any query”

Reply:
(because otherwise it can be read as possibly canceling some other query instance if running on the specified thread)

Reply:
Perhaps "Method to cancel an execution of this query in the parameter thread."